src/components/reddit-feed.tsx

```diff
@@ -23,15 +25,19 @@ export default function RedditFeed() {
   }>({});
 
   function getRedditPosts() {
-    // TODO -move this to a serverless function in the /api folder and call it from the client
-    fetchRedditPosts()
+    fetch(REDDIT_GET_URL)
+      .then((response) => {
+        if (response.ok) return response.json();
+        else throw (response.statusText)
+      })
       .then((response) => {
         setRedditPosts(response);
         setLoading(false);
       })
       .catch((error) => {
         console.error("Failed to fetch reddit posts", error);
-      });
+        setTimeout(() => getRedditPosts(), TIMEOUT_RETRY);
```

Comment:
Does this mean that it will keep retrying in the case that the posts do not load? I think part of the reason they are not loading is likely due to API key limits, but I haven't investigated it throughly.

I'm ok with a retry or 2, but don't want it trying to refecth on a loop. I would rather just show the skelton in the 'loading' state.

Reply:
@JeanPec - let me know about this change. Would like to get it out there this weekend.

Reply:
Okay I will add a count I was not sure if I should do an infinite refresh. Yes we can display the loading if the user really want to get rid of it he can refresh his page

Reply:
@erin-fitzpatric I just added a limit to the number of retry 👍